### DISCUSSION TITLE
Looking for a maintainer

### DISCUSSION THREAD

**Discussion by spaze** [spaze/phpstan-stripe#19] [General]
I (@spaze) have stopped working on a project that uses the Stripe PHP SDK and unfortunately I'm not currently using Stripe in anyway so making changes (and verifying them in real-world tests!) became a bit difficult.

And because I don't want to rely on feedback only and because I like dogfooding, I'd like to invite someone else to co- or even fully (preferred) maintain the extension. Let me know if you'd like to carry on.

I'll keep the repo available and will merge PRs until the end of 2023 and then archive the repository and won't publish any new releases unless of course I'll find a maintainer, or I'll start using Stripe again.

Thanks everyone using the extension, makes me happy to see it helps one way or another.

**Comment by anthonyryan1** [comment]
I'm in the process of integrating PHPStan and this extension into our regular testing process at my work.

When that's complete we would have real world usage of charge, paymentintent, subscription & customer objects, which is probably the majority of popular functionality.

Given I'm still relatively new to PHPStan, I'm not comfortable taking over maintainership yet. But if nobody else comes forward I'm happy to do so in 4-6 months. By that point I think I'll be familiar enough to not make a mess of things.

**Comment by spaze** [comment]
Hey @turbo124, I've noticed you're using this PHPStan extension for [Invoice Ninja](https://github.com/invoiceninja/invoiceninja) so I just wanted to make sure you know about the planned EOL, see the top post here for reasons.

I'm reaching out like this because I've noticed a [spike in installs](https://packagist.org/packages/spaze/phpstan-stripe/stats) and it most probably could be attributed to you [adding it](https://github.com/invoiceninja/invoiceninja/pull/8711) and I just don't want to kill it like that, silently :-) If you'd like to take it over, I'd very happy (of course unless @anthonyryan1 & PHPStan are much better friends now :-) )

**Comment by anthonyryan1** [comment]
I'm willing to take it over before letting it die! But if anyone has more active interest I'm open to co-maintaining as well.

**Pros**
- I've added PHPStan to the CI pipeline for 2/3 of the major PHP codebases I'm responsible.
- I will be continuing to PHPStan lint a codebase using this for the foreseeable future.
- I've also developed a a PHPStan extensions  in the style as this for another API, because I found this one beneficial (sadly not an open source API).

**Cons**
- I have no particular vision for a direction to take this other than fixing bugs and compatibility with newer PHP versions, etc.
- Work keeps me rather busy, so I would probably only be working on this a few times a year and when issues occur or are reported.

With all my cards on the table. If you still feel like I'm the best (or only) option to continue maintenance, I'll happily do so.

**Reply by turbo124** [reply]
Happy to support @anthonyryan1 take the lead. 

My card is pretty full, but Stripe is a big part of what we do so I'll be happy to contribute where I can.

**Reply by spaze** [reply]
Thanks @anthonyryan1 & @turbo124! Very happy that you're willing to take over and/or support it :-)

~~I'll now probably create an organization and transfer the repo there, probably `https://github.com/phpstan-stripe/phpstan-stripe`, and then I'll add both of you to the organization. Or I can transfer it to a user, which woudl be one of you, but I'd rather go with an organization, it makes it easier to transfer ownerships in the future. I'll see what I can about Packagist package.~~ No need to overcomplicate things for now.

And @anthonyryan1, your _pros_ are very cool, good job! And those _cons_? This is **exactly** how much time I was dedicating to it and how I felt about this package. I believe this packages requires just maintenance mostly, check and merge half a PR/year in average.

My ideas (can't even call it a vision or direction):
1. I had an idea to convert this to [PHPStan stub files](https://phpstan.org/user-guide/stub-files) but I have done zero research on it, whether it would bring us something new or useful. The way it works now is kinda similar anyway, and it was working well enough for me and my case that it was just an idea, maybe even just a shower thought.

2. Right now, this doesn't support the latest stripe/stripe-php versions, I believe the latest is 13.something, while this package only supports 10.x, so make it work with the newer versions would be nice:
https://github.com/spaze/phpstan-stripe/blob/ce764d07dbdfe175e9af0aa8ce3dd5c665ad880a/composer.json#L32

**Reply by anthonyryan1** [reply]
I appreciate your ideas. It's a good note on where we're at right now.

I have no preference on moving it to an organization or person, but I'll plan some time in the next two weeks to get this compatible with the newest `stripe/stripe-php`.

**Reply by spaze** [reply]
I have added you both to this repo as collaborators. I think the move to an organization can wait a bit, no need to overcomplicate things, at least not now. I don't want anyone to have to just go and change their dependencies because of the move.

**Reply by mattvb91** [reply]
Just wanted to say thank you guys for maintaining. Not using this yet but looks like its exactly what i need for upcoming project.

**Reply by Bellangelo** [reply]
Have you made any progress on this? I can give a hand as well. Although, my time is kind of limited right now.

**Reply by spaze** [reply]
Yeah @Bellangelo, the new version was released by @turbo124 already: https://github.com/spaze/phpstan-stripe/releases/tag/v3.2.0
